util/ranger/detacher.go

```diff
@@ -304,7 +347,16 @@ func (d *rangeDetacher) detachCNFCondAndBuildRangeForIndex(conditions []expressi
 				return &DetachRangeResult{}, nil
 			}
 			if len(tailRes.AccessConds) > 0 {
-				res.Ranges = appendRanges2PointRanges(res.Ranges, tailRes.Ranges)
+				for i, resRange := range res.Ranges {
+					var saveRange []*Range
+					saveRange = append(saveRange, resRange)
+					if len(resRange.LowVal) == eqOrInCount {
+						saveRange = appendRanges2PointRanges(saveRange, tailRes.Ranges)
+						res.Ranges[i] = saveRange[0]
```

Comment:
Why only the first element in `saveRange`?

Reply:
fixed